cypress/e2e/main.cy.js

```diff
@@ -1,36 +1,20 @@
 describe("Main page", () => {
   beforeEach(() => {
     cy.visit("http://localhost:3000");
+    cy.intercept({
+      method: 'GET',
+      url: 'https://rancid-tomatillos.herokuapp.com/api/v2/movies'
+    }, { fixture: 'movies' })
   });
   describe("When main page render correctly", () => {
     it("Should render correctly", () => {
-      cy.url().should("include", "3000");
-      cy.contains("a");
-      cy.get(".poster-display").find(".main-board").should("have.length", 40);
+      cy.get('.poster-display > :nth-child(1)')
+    
+      cy.get(".poster-display").find(".main-board").should("have.length", 3);
     });
-
     it("When user click movie card, it should have the correct URL", () => {
-      const sampleData = {
-        movies: [
-          {
-            id: 436270,
-            poster_path:
-              "https://image.tmdb.org/t/p/original//pFlaoHTZeyNkG83vxsAJiGzfSsa.jpg",
-            backdrop_path:
-              "https://image.tmdb.org/t/p/original//bQXAqRx2Fgc46uCVWgoPz5L5Dtr.jpg",
-            title: "Black Adam",
-            average_rating: 4,
-            release_date: "2022-10-19",
-          },
-        ],
-      };
-      cy.intercept(
-        "get",
-        "https://rancid-tomatillos.herokuapp.com/api/v2/movies",
-        sampleData
-      )
-        .get("#root > div > div.poster-display > div > a")
-        .should("have.attr", "href", `/details/${sampleData.movies[0].id}`);
+      cy.get("#root > div > div.poster-display > div > a")
```

Comment:
I understand why we can use a fixture here, I think a benefit of having the mock data inside the test itself makes the test self-contain and not sensitive to any data change.